build_mt.py

```diff
@@ -1,599 +1,827 @@
-cat_lc_dir = "/mnt/gosling1/boryanah/light_cone_catalog/"+sim_name+"/halos_light_cones/"
-if not os.path.exists(cat_lc_dir): os.makedirs(cat_lc_dir)
-
-# directory where we save the current state if we want to resume 
-if not os.path.exists(os.path.join(cat_lc_dir,'tmp')): os.makedirs(os.path.join(cat_lc_dir,'tmp'))
-resume = False
-                      
-# all redshifts, steps and comoving distances of light cones files; high z to low z
-zs_all = np.load("data_headers/redshifts.npy")
-chis_all = np.load("data_headers/coord_dist.npy")
-zs_all[-1] = float('%.1f'%zs_all[-1])
-
-# get functions relating chi and z
-chi_of_z = interp1d(zs_all,chis_all)
-z_of_chi = interp1d(chis_all,zs_all)
-
-# more accurate, slightly slower
-if not os.path.exists("data/zs_mt.npy"):
-    # all merger tree snapshots and corresponding redshifts
-    snaps_mt = sorted(glob.glob(merger_dir+"associations_z*.0.asdf"))
-    print(snaps_mt)
-    zs_mt = get_zs_from_headers(snaps_mt)
-    np.save("data/zs_mt.npy",zs_mt)
-zs_mt = np.load("data/zs_mt.npy")
-
-
-# fields are we extracting from the merger trees
-#fields_mt = ['HaloIndex','HaloMass','Position','MainProgenitor','Progenitors','NumProgenitors']
-# lighter version
-fields_mt = ['HaloIndex','Position','MainProgenitor']
-
-# redshift of closest point on wall between original and copied box
-z1 = z_of_chi(0.5*Lbox-origin[0])
-# redshift of closest point where all three boxes touch
-#z2 = z_of_chi((0.5*Lbox-origin[0])*np.sqrt(2))
-# furthest point where all three boxes touch; TODO: I think that's what we need
-z2 = z_of_chi((0.5*Lbox-origin[0])*np.sqrt(3))
-
-# corresponding indices
-ind_start = np.argmin(np.abs(zs_mt-z_start))
-ind_stop = np.argmin(np.abs(zs_mt-z_stop))
-
-# initialize difference between the conformal time of last two shells
-delta_chi_old = 0.
-
-# loop over each merger tree redshift
-for i in range(ind_start,ind_stop+1):
-    
-    # this snapshot and previous
-    z_this = zs_mt[i]
-    z_prev = zs_mt[i+1]
-    print("redshift of this and previous snapshot = ",z_this,z_prev)
-    
-    # up to z1, we work with original box
-    if z_this < z1: copies_this = 1
-    # after z2, we work with 2 copies of the box
-    elif z_this > z2: copies_this = 2
-    # between z1 and z2, we work with 3 copies of the box
-    else: copies_this = 3
-
-    # up to z1, we work with original box
-    if z_prev < z1: copies_prev = 1
-    # after z2, we work with 2 copies of the box
-    elif z_prev > z2: copies_prev = 2
-    # between z1 and z2, we work with 3 copies of the box; could be improved needs testing
-    else: copies_prev = 3
-
-    print("copies of the box needed for this and previous snapshot = ",copies_this,copies_prev)
-    
-    # number of copies should be same and equal to max of the two; repetitive
-    copies_this = np.max([copies_this,copies_prev])
-    copies_prev = np.max([copies_this,copies_prev])
-    
-    # previous redshift, distance between shells and copies
-    if resume == True:
```

Comment:
In general, use the truthiness of Python variables when possible!  Like `if resume:` instead of `if resume == True:`